fendermint/vm/interpreter/src/fvm/checkpoint.rs

```diff
@@ -63,30 +63,11 @@ where
         .block_hash()
         .ok_or_else(|| anyhow!("block hash not set"))?;
 
-    let batch = gateway.bottom_up_msg_batch(state, height.into())?;
-
-    if batch.block_height.as_u64() == 0 {
-        tracing::debug!(height = height.value(), "no bottom up msg batch at height");
-
-        // we need to check if the checkpoint period has reached
-        if should_create_checkpoint(gateway, state, height)?.is_none() {
-            tracing::debug!(
-                height = height.value(),
-                "not bottom up checkpoint period height"
-            );
-            return Ok(None);
-        }
-
-        tracing::debug!(
-            height = height.value(),
-            "bottom up checkpoint period height reached"
-        );
+    let (msgs, subnet_id) = if let Some(v) = should_create_checkpoint(gateway, state, height)? {
+        v
     } else {
-        tracing::debug!(
-            height = height.value(),
-            "bottom up msg batch found at height"
-        );
-    }
+        return Ok(None);
+    };
```

Comment:
You can use the new [let-else](https://doc.rust-lang.org/rust-by-example/flow_control/let_else.html) syntax here:

```rust
let Some((msgs, subnet_id)) = should_create_checkpoint(...) else {
  return Ok(None);
}
```